docarray/doc_index/backends/weaviate_doc_index.py

```diff
@@ -0,0 +1,594 @@
+
+                batch.add_data_object(
+                    uuid=weaviate.util.generate_uuid5(parsed_doc, index_name),
+                    data_object=parsed_doc,
+                    class_name=index_name,
+                    vector=vector,
+                )
+
+    def _text_search(self, query: str, search_field: str, limit: int) -> _FindResult:
+        index_name = self._db_config.index_name
+        bm25 = {"query": query, "properties": [search_field]}
+
+        results = (
+            self._client.query.get(index_name, self.properties)
+            .with_bm25(bm25)
+            .with_limit(limit)
+            .with_additional(["score", "vector"])
+            .do()
+        )
+
+        return self._format_response(results["data"]["Get"][index_name], "score")
+
+    def _text_search_batched(
+        self, queries: Sequence[str], search_field: str, limit: int
+    ) -> _FindResultBatched:
+        qs = []
+        for i, query in enumerate(queries):
+            bm25 = {"query": query, "properties": [search_field]}
+
+            q = (
+                self._client.query.get(self._db_config.index_name, self.properties)
+                .with_bm25(bm25)
+                .with_limit(limit)
+                .with_additional(["score", "vector"])
+                .with_alias(f'query_{i}')
+            )
+
+            qs.append(q)
+
+        results = self._client.query.multi_get(qs).do()
+
+        docs_and_scores = [
+            self._format_response(result, "score")
+            for result in results["data"]["Get"].values()
+        ]
+
+        docs, scores = zip(*docs_and_scores)
+        return list(docs), list(scores)
+
+    def execute_query(self, query: Any, *args, **kwargs) -> Any:
+        da_class = DocumentArray.__class_getitem__(
+            cast(Type[BaseDocument], self._schema)
+        )
+
+        if isinstance(query, self.QueryBuilder):
+            batched_results = self._client.query.multi_get(query._queries).do()
+            batched_docs = batched_results["data"]["Get"].values()
+
+            def f(doc):
+                return self._schema.from_view(self._parse_weaviate_result(doc))
+
+            results = [
+                da_class([f(doc) for doc in batched_doc])
+                for batched_doc in batched_docs
+            ]
+            return results if len(results) > 1 else results[0]
+
+    def num_docs(self) -> int:
+        index_name = self._db_config.index_name
+        result = self._client.query.aggregate(index_name).with_meta_count().do()
+        # TODO: decorator to check for errors
+        total_docs = result["data"]["Aggregate"][index_name][0]["meta"]["count"]
+
+        return total_docs
+
+    def python_type_to_db_type(self, python_type: Type) -> Any:
+        """Map python type to database type."""
+        for allowed_type in WEAVIATE_PY_VEC_TYPES:
+            if issubclass(python_type, allowed_type):
+                return np.ndarray
```

Comment:
So linking back to my comment above, this should return the weaviate specific column type, not `np.ndarray`. Maybe this is not well explained in the manual, because in the ES implementation the same thing happened in the beginning...